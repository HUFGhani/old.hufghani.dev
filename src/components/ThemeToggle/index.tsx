import { MoonIcon, SunIcon } from '@chakra-ui/icons'
import { IconButton, keyframes, useColorMode, useColorModeValue } from '@chakra-ui/react'
import React from 'react'

const ThemeToggle: React.FC = () => {
  const { toggleColorMode: toggleMode } = useColorMode()
  const ToggleIcon = useColorModeValue(SunIcon, MoonIcon)
  const spin = keyframes`
  from {transform: rotate(0deg);}
  to {transform: rotate(360deg)}
`
  const spinAnimation = `${spin} infinite 2s linear`

  return (
    <IconButton
      icon={
        <ToggleIcon
          sx={{
            _hover: {
              animation: spinAnimation,
            },
          }}
        />
      }
      variant="ghost"
      aria-label="Toggle Theme"
      onClick={toggleMode}
      size={`md`}
      mx="0"
    />
  )
}

export default ThemeToggle
