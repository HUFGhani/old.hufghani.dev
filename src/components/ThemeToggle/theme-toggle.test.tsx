import { ChakraProvider, CSSReset } from '@chakra-ui/react'
import { render } from '@testing-library/react'
import React from 'react'
import ThemeToggle from '.'

const renderThemeToggle = () => {
  return render(
    <ChakraProvider>
      <CSSReset />
      <ThemeToggle />
    </ChakraProvider>
  )
}

describe(`<ThemeToggle/>`, () => {
  it(`should render ThemeToggle`, () => {
    const { asFragment } = renderThemeToggle()
    expect(asFragment()).toMatchSnapshot()
  })
})
