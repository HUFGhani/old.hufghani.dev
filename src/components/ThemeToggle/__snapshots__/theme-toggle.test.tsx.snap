// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ThemeToggle/> should render ThemeToggle 1`] = `
<DocumentFragment>
  <button
    aria-label="Toggle Theme"
    class="chakra-button css-150nmn8"
    type="button"
  >
    <svg
      aria-hidden="true"
      class="chakra-icon css-1q8md0w"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <g
        fill="none"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
      >
        <circle
          cx="12"
          cy="12"
          r="5"
        />
        <path
          d="M12 1v2"
        />
        <path
          d="M12 21v2"
        />
        <path
          d="M4.22 4.22l1.42 1.42"
        />
        <path
          d="M18.36 18.36l1.42 1.42"
        />
        <path
          d="M1 12h2"
        />
        <path
          d="M21 12h2"
        />
        <path
          d="M4.22 19.78l1.42-1.42"
        />
        <path
          d="M18.36 5.64l1.42-1.42"
        />
      </g>
    </svg>
  </button>
</DocumentFragment>
`;
